src/plugins/vis_type_timeseries/public/types.ts

```diff
@@ -20,10 +20,40 @@
 import React from 'react';
 import { EuiDraggable } from '@elastic/eui';
 
+import type { PANEL_TYPES } from '../common/panel_types';
+import { TimeseriesUIRestrictions } from '../common/ui_restrictions';
+
 type PropsOf<T> = T extends React.ComponentType<infer ComponentProps> ? ComponentProps : never;
 type FirstArgumentOf<Func> = Func extends (arg1: infer FirstArgument, ...rest: any[]) => any
   ? FirstArgument
   : never;
 export type DragHandleProps = FirstArgumentOf<
   Exclude<PropsOf<typeof EuiDraggable>['children'], React.ReactElement>
 >['dragHandleProps'];
+
+/**
+ * series data is not typed yet.
+ * Should be in sync with returned data from server [GetVisDataPanel] in src/plugins/vis_type_timeseries/server/lib/get_vis_data.ts
+ */
+interface SeriesData {
```

Comment:
maybe we should move that types into common folder like we already did for payload types?  I think BE return exactly the same types which should be reused it future

Reply:
Yes, I totally agree with you. That's why I left a comment here. But I believe this could be done in scope of https://github.com/elastic/kibana/issues/63593

Reply:
I hope so, but do we have any reason not to do this now? I'm ok see the non-detailed types for this iteration of work.

Reply:
Moved into common folder